lib/jekyll/collection.rb

```diff
@@ -95,14 +95,13 @@ def filtered_entries
         end
     end
 
-    # The directory for this Collection, relative to the site source.
+    # The directory for this Collection, relative to the site source or the directory
+    # containing the collection.
     #
     # Returns a String containing the directory name where the collection
     #   is stored on the filesystem.
     def relative_directory
-      @relative_directory ||= Pathname.new(directory).relative_path_from(
-        Pathname.new(site.source)
-      ).to_s
+      @relative_directory ||= "_#{label}"
```

Comment:
This should be relative to the site source, so if source is `.` then it should be `collections_dir/_label`.

Reply:
Things that're going to break **if `@relative_directory ||= collections_dir/_label`**:
https://github.com/jekyll/jekyll/blob/f77d704a1b97ac43684df85e92a8738412d1725a/lib/jekyll/document.rb#L32-L36
https://github.com/jekyll/jekyll/blob/f77d704a1b97ac43684df85e92a8738412d1725a/lib/jekyll/document.rb#L76-L79
https://github.com/jekyll/jekyll/blob/f77d704a1b97ac43684df85e92a8738412d1725a/lib/jekyll/document.rb#L121-L124